source/xatlas/xatlas.cpp

```diff
@@ -6771,11 +6899,13 @@ struct PiecewiseParam
 			}
 			XA_DEBUG_ASSERT(freeVertex != UINT32_MAX);
 			if (m_vertexInPatch.get(freeVertex)) {
-#if 0
+// #if 0
 				// If the free vertex is already in the patch, the face is enclosed by the patch. Add the face to the patch - don't need to assign texcoords.
+				if (m_faceInAnyPatch.get(oface))
+					continue;
```

Comment:
just this part if @jpcy can give an opinion he'd be the most relevant person for analysis.

Reply:
@siliconvoodoo  yeah, I understand your concern. I've added this check because a triangle/face can be added multiple times to an array. But sure, @jpcy knows better if it makes sense.
https://github.com/jpcy/xatlas/blob/f01768abf0358d47e116755b107a89ac888f8614/source/xatlas/xatlas.cpp#L6879

Reply:
As a variant we could leave only one check for the sake of performance. @siliconvoodoo @castano what do you think about it?
![image](https://github.com/user-attachments/assets/e700ecd9-8689-4053-b5e6-a5f13c13c0f1)

Reply:
Apologies I've deleted my last comment because there were different checks.

Reply:
Ah, right! I'll delete this check with a new commit. 
![image](https://github.com/user-attachments/assets/bb260719-529e-4636-9ec3-667232604601)